File: backend/discounts/models/product.go
package models

import (
	"github.com/satori/go.uuid"
)

type Product struct {
	tableName    struct{}  `sql:"product"`
	ID           uuid.UUID `json:"id", sql:"type:uuid"`
	PriceInCents uint64    `json:"price_in_cents"`
	Title        string    `json:"title"`
	Description  string    `json:"description"`
}

File: backend/products/src/models/Product.ts
import { Entity, Column, PrimaryColumn } from "typeorm";
import { Discount } from "./Discount";

@Entity('product')
export class Product {

    @PrimaryColumn('uuid')
    id!: string;

    @Column('int')
    price_in_cents!: number;

    @Column('text')
    title!: string;

    @Column('text')
    description!: string;

    discount: Discount = { percentage: 0, value_in_cents: 0}

    setDiscount(percentage: number, value_in_cents: number){
        this.discount = {
            percentage,
            value_in_cents
        };
    }
}

